runtime/port/sysvipc/j9shmem.c

```diff
@@ -1419,6 +1419,24 @@ j9shmem_createDir(struct J9PortLibrary* portLibrary, char* cacheDirName, uintptr
 		rc = createDirectory(portLibrary, (char*)pathBuffer, cacheDirPerm);
 		if (J9SH_FAILED != rc) {
 			if (J9SH_DIRPERM_ABSENT == cacheDirPerm) {
+				char *cacheParentDirPtr = strstr(cacheDirName, J9SH_HIDDENDIR);
```

Comment:
By default, the cache directory is inside user's home, see https://github.com/eclipse-openj9/openj9/blob/c9cb232d5837f091c6cf2b07615bee022da17698/runtime/port/sysvipc/j9shmem.c#L1274

The user was complaining about the 777 permission of the parent directory inside the user's home directory. User's home directory is not supposed to be shared with other users so 777 does not make sense.  We only need to change the permission of the parent directory if it is inside user home. 

Similarly to how we set and use `usingDefaultTmp` in this function, you could have a local variable `insiderUserHome` and only change the parent directory permission when it is TRUE.